ragna/source_storages/_chroma.py

```diff
@@ -36,19 +36,26 @@ def __init__(self) -> None:
             )
         )
 
-    def _get_collection(self) -> chromadb.Collection:
+    def _get_collection(self, corpus_name: Optional[str]) -> chromadb.Collection:
+        from chromadb.api.types import EmbeddingFunction as ChromaEmbeddingFunction
+
+        if corpus_name is None:
+            corpus_name = self._embedding_id
+
         return self._client.get_or_create_collection(
-            self._embedding_id, embedding_function=self._embedding_function
+            corpus_name,
+            embedding_function=cast(ChromaEmbeddingFunction, self._embedding_function),
```

Comment:
Is the `cast` required here? We are already doing that when setting the attribute

https://github.com/Quansight/ragna/blob/7c5728af3673868222ded6f06bb02f6f6d9cf03a/ragna/source_storages/_vector_database.py#L71-L74